components/datetime/src/pattern/mod.rs

```diff
@@ -62,6 +84,15 @@ impl Pattern {
             .parse_placeholders(vec![time, date])
             .map(Self)
     }
+
+    /// Returns the maximum time granularity contained in the pattern,
+    /// or `None` if the pattern does not contain time.
+    /// e.g. The maximum granularity for `h:mm` is `Minutes`.
+    /// e.g. The maximum granularity for `h:mm:ss` is `Seconds`.
+    /// e.g. The maximum granularity for `E, dd/MM/y` is `None`.
+    pub fn most_granular_time(&self) -> Option<TimeGranularity> {
+        self.0.iter().flat_map(PatternItem::time_granularity).max()
+    }
```

Comment:
Same comment as above. I wouldn't put such specialized methods on `PatternItem`. Maybe even add a `relative_time` module with helper functions for calculations around it.

It's purely a recommendation for code readability.